src/libsystemd/sd-bus/bus-internal.c

```diff
@@ -91,8 +91,10 @@ bool interface_name_is_valid(const char *p) {
                                 (!dot && *q >= '0' && *q <= '9') ||
                                 *q == '_';
 
-                        if (!good)
+                        if (!good) {
+                                log_debug("The interface %s is invalid as it contains special character %c", p, *q);
```

Comment:
I'm not sure if we should include such debug helpers in general code.
But even ignoring the overall desirability, you don't know whether the `p` string is valid printable non-cc unicode, so you cannot include it in a debug message.

Reply:
So, you mean just print the invalid special char?

Reply:
@ya1gaurav see Lennart's comment about cescape() https://github.com/systemd/systemd/pull/17857#issuecomment-741915564